### PR TITLE
Add type annotations to distribution.py

### PR DESCRIPTION
**Pull request by EPronovost**
As title.

### CODE REVIEW

**Review by kit1980 — APPROVED** [review]

**Comment by kit1980** [comment]
@EPronovost The linter says the types are not correct.

**Review by kit1980 — COMMENTED** [review]
Please fix lint issues.

**Comment by EPronovost** [comment]
Hi @kit1980 , thanks for the review. The github action bot says I should add release notes and topic labels to the PR, but the github UI says I don't have permission to add labels to PRs. Is there something I should do here?

**Comment by kit1980** [comment]
@EPronovost It's fine ignore it for now. You can add labels by commenting to pytorchbot, see https://github.com/pytorch/pytorch/wiki/Bot-commands#labeling

File: torch/distributions/distribution.py [hunk 1]
```diff
@@ -116,58 +122,58 @@ def support(self) -> Optional[Any]:
...
+    def sample(self, sample_shape: torch.Size = torch.Size()) -> torch.Tensor:
         """
         Generates a sample_shape shaped sample or sample_shape shaped batch of
         samples if the distribution parameters are batched.
         """
         with torch.no_grad():
             return self.rsample(sample_shape)
 
-    def rsample(self, sample_shape=torch.Size()):
+    def rsample(self, sample_shape: torch.Size = torch.Size()) -> torch.Tensor:
```
Comment (pmeier) [inline torch/distributions/distribution.py:160]: @EPronovost @kit1980 

`torch.Size` is too strict here since something like

```py
dist = torch.distributions.Uniform(0.0, 1.0)
dist.sample((2, 3))
```

is no longer allowed since `(2, 3)` is not a `torch.Size`. This forces users to do 

```py
dist.sample(torch.Size((2, 3)))
```

which is likely not intended, is it? We should use `torch.types._size`

https://github.com/pytorch/pytorch/blob/166b5d3e7c5c230c455dcbcc05c84dd6bc03721b/torch/types.py#L22

as annotation here and in all other occurrences of this annotation.

Comment (kit1980) [reply torch/distributions/distribution.py:160]: @pmeier Thanks! Do you want to send a PR?
